core/components/catalyst-image/index.tsx

```diff
@@ -0,0 +1,33 @@
+'use client';
+
+// eslint-disable-next-line @typescript-eslint/no-restricted-imports
+import Image, { ImageProps } from 'next/image';
+
+import bcCdnImageLoader from '~/lib/cdn-image-loader';
+
+const cdnHostname = process.env.BIGCOMMERCE_CDN_HOSTNAME ?? 'cdn11.bigcommerce.com';
+
+function shouldUseLoaderProp(props: ImageProps): boolean {
+  return typeof props.src === 'string' && props.src.includes(cdnHostname);
```

Comment:
🍹 would a startsWith check be slightly safer?

Reply:
If we want to go super safe `new URL(props.src, 'http://n').hostname === cdnHostname` could do the trick. We'd also probably want to try/catch `new URL` in case an invalid URL is passed in.

Doing `startsWith(`https://${cdnHostname}`) is probable the sane thing to do

Reply:
I wanted to avoid `startsWith` but I guess we should. Went with that instead of `new URL`.